src/libutil/include/nix/util/logging.hh

```diff
@@ -196,7 +206,20 @@ extern std::unique_ptr<Logger> logger;
 
 std::unique_ptr<Logger> makeSimpleLogger(bool printBuildLogs = true);
 
-std::unique_ptr<Logger> makeJSONLogger(Descriptor fd);
+/**
+ * Create a logger that sends log messages to `mainLogger` and the
+ * list of loggers in `extraLoggers`. Only `mainLogger` is used for
+ * writing to stdout and getting user input.
+ */
```

Comment:
thought, no action required:
Would be nice to split these concerns at some point. A UI is a logger, but a logger is not a UI, something like that.